### ISSUE TITLE
ghc/ghcjs 8.6

### ISSUE THREAD

**Issue by tysonzero** [NixOS/nixpkgs#172802]
**Project description**
ghcjs-8.10 seems more or less broken for large projects due to massive memory usage and performance regression, as well as a regular bug: https://github.com/ghcjs/ghcjs/issues/821

Adding support for ghcjs-8.6 along with re-adding ghc-8.6 to go with it would allow us to move to a new nixpkgs which would be fantastic.

**Metadata**

* homepage URL: https://github.com/ghcjs/ghcjs/tree/ghc-8.6
* source URL:https://github.com/ghcjs/ghcjs/tree/ghc-8.6
* license: mit
* platforms: unix, linux, darwin, windows

**Comment by cdepillabout** [comment]
cc @dfordivam and @Ericson2314

**Comment by dfordivam** [comment]
@tysonzero Using ghc 8.6 on the latest nixpkgs would not work smoothly / at all, there would be a number of haskell packages which would fail to compile. And I dont think the haskell package maintainers would be willing to add the ghc 8.6 back. In the current situation the ideal solution is to get the ghcjs 8.10 fixed.

Nevertheless you can still use ghc/ghcjs 8.6 in your projects by including an older nixpkgs, just for the ghc/ghcjs, and then doing an override of the ghc/ghcjs in the latest nixpkgs. Of course things would be quite broken in my opinion.

**Comment by sternenseemann** [comment]
I think packaging GHCJS 8.6 would have the following prerequisites:

* buildable using (largely at least) Stackage LTS 19, downgrades are possible, but painful / a lot of manual effort
* should be buildable by Hydra, i.e. ghcjs 8.6 + core packages should be less than 3GB in size. This is not the case for ghcjs 8.10.7 which makes it painful to maintain because easy to fix regressions don't get caught until it breaks a downstream user's setup

**Comment by tysonzero** [comment]
@dfordivam Yeah older nixpkgs are getting rather difficult to use with time. Particularly with the `git://` deprecation basically obliterating all old nixpkgs versions that don't cache hit, wish we had nix/ipfs already.
